lib/scanners/base.js

```diff
@@ -34,17 +34,41 @@ var BaseScanner = function (beaconMapper, beaconRegionMapper, beaconRegionMonito
     this.init = function (networkIdentifier) {
         if (!_.isUndefined(networkIdentifier)) {
             self.networkId = networkIdentifier;
+            var hasPermissions = function () {
+                beaconHandler.init();
+                beaconLog.init();
+                self.setBeaconRegions([]);
+                if (OS_ANDROID || (OS_IOS && (_.isUndefined(Ti.App.arguments.launchOptionsLocationKey) || Ti.App.arguments.launchOptionsLocationKey === false))) {
+                    knownBeaconService.refreshBeacons([self.networkId]);
+                }
+            };
+
             // Request authorization is required on IOS devices
             if (OS_IOS) {
-                Ti.Geolocation.requestAuthorization(Ti.Geolocation.AUTHORIZATION_ALWAYS);
-            }
-
-            beaconHandler.init();
-            beaconLog.init();
-            self.setBeaconRegions([]);
-            if (OS_ANDROID || (OS_IOS && (_.isUndefined(Ti.App.arguments.launchOptionsLocationKey) || Ti.App.arguments.launchOptionsLocationKey === false))) {
-                knownBeaconService.refreshBeacons([self.networkId]);
+                var version = Ti.version.split(".");
+                var permissionType = Ti.Geolocation.AUTHORIZATION_ALWAYS;
+                if (version[0] < 5 || // Version < 5
+                    (version[0] === 5 && version[1] === 0) // Version 5.0.*
+                ) {
+                    // BC: request permission the old way
+                    Ti.Geolocation.requestAuthorization(Ti.Geolocation.AUTHORIZATION_ALWAYS);
+                    hasPermissions();
+                    return;
+                }
+
+                if (Ti.Geolocation.hasLocationPermissions(permissionType)) {
+                    hasPermissions();
+                    return;
+                }
+
+                Ti.Geolocation.requestLocationPermissions(permissionType, function(res) {
+                    if (res.success) {
+                        hasPermissions();
+                    }
+                });
+                return;
```

Comment:
Isn't it possible to split the entire content of this `if` to a separate function? It's rather hard to read the control structure here, if you ask me.